src/generators/Silk.NET.SilkTouch.Symbols/StructSymbol.cs

```diff
@@ -9,11 +9,31 @@ namespace Silk.NET.SilkTouch.Symbols;
 /// A <see cref="TypeSymbol"/> representing a <c>struct</c>.
 /// </summary>
 /// <param name="Identifier">The Identifier of this struct</param>
-/// <param name="Members">The Members of this struct</param>
+/// <param name="Layout">The layout of this struct</param>
 /// <remarks>
 /// In this context, a Struct means a type that represents the layout of a continuous block of memory.
 /// </remarks>
 // /// Each meaningful place in this memory called a field (see <see cref="FieldSymbol"/>) is accessible via this type.
 // /// Fields are allowed to overlap.
 // /// Additionally it may contain one or multiple <see cref="MethodSymbol"/> that are called with an instance of this type as their first argument.
-public sealed record StructSymbol(IdentifierSymbol Identifier, ImmutableArray<MemberSymbol> Members) : TypeSymbol(Identifier);
+public sealed record StructSymbol(IdentifierSymbol Identifier, StructLayout Layout) : TypeSymbol(Identifier);
+
+/// <summary>
+/// A <see cref="StructSymbol"/> representing the layout of a <see cref="StructSymbol"/>
+/// </summary>
+/// <param name="Entries">The entries of this layout</param>
+public sealed record StructLayout(ImmutableArray<LayoutEntry> Entries)
+{
+    /// <summary>
+    /// An empty layout with no members
+    /// </summary>
+    public static readonly StructLayout Empty = new StructLayout(ImmutableArray<LayoutEntry>.Empty);
+}
+
+/// <summary>
+/// Represents an entry in a <see cref="StructLayout"/>
+/// </summary>
+/// <param name="Member">The member symbol associated with this layout entry</param>
+/// <param name="ByteOffset">The offset of this entry in bytes</param>
+/// <seealso cref="StructLayout"/>
+public sealed record LayoutEntry(MemberSymbol Member, int ByteOffset);
```

Comment:
ByteOffset should probably be nullable in cases where we want the runtime to figure it out.

Reply:
Do we ever want that? I've just glanced at Clangsharp, I thought that always tells us where exactly the field should go?

Reply:
I'm not sure if the XML does all of the time, but ClangSharp always has that data available to it internally at least. In any case, ClangSharp by default uses implicit layout in the bindings it generates unless it's necessary to be explicit. In cases where it's explicit, it is always present in the XML.

Reply:
OK, then it's reasonable to include a sequential layout. I'll open up a separate PR for that though, if that's OK. Trying to get as good of a git history as possible

Reply:
fine

Reply:
(Work Item 69)

Reply:
(nice)